app/controllers/admins/orders_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Admins
+  class OrdersController < AdminsController
+    def index
+      @orders = Order.order(delivery_status: :desc)
+    end
+
+    def update
+      @order = Order.find(params[:id])
+      if @order.delivered?
+        redirect_to admins_orders_url, notice: 'You can\'t deliver this order again.'
```

Comment:
I would change single quotation mark to double. With double we don't have to use escape characters for single quotations, and its more readable. But this is just small sugesstion.
```suggestion
        redirect_to admins_orders_url, notice: "You can't deliver this order again."
```

Reply:
done :)